### PR TITLE
feat: Http handlers & Dad Jokes

### PR DESCRIPTION
**Pull request by Soap-141**
GitHub Issue: #

## PR Type

What kind of change does this PR introduce?

- Feature
- Code style update (formatting)
- Refactoring (no functional changes, no api changes)

## Description

Added Dad Jokes page.
Replaced Xamarin Essentials Connectivity by an Uno implementation.
Enabled some API configuration.

## PR Checklist 
Please check if your PR fulfills the following requirements:

- [x] Interface members are XML documented
- [x] Documentation (XML or comments) has been added and/or existing documentation has been updated
- [ ] [Architecture documents](./doc/Architecture.md) have been updated
- [x] Tested on all relevant platforms

### CODE REVIEW

File: src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj [hunk 1]
```diff
@@ -11,6 +11,7 @@
 		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-ios'">14.2</SupportedOSPlatformVersion>
 	</PropertyGroup>
 	<ItemGroup>
+		<PackageReference Include="Chinook.DataLoader.Uno.WinUI" Version="1.1.0" />
```
Comment (vincentcastagna) [inline src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:14]: Why was this added on this PR ? Were we missing it ?

Comment (Soap-141) [reply src/app/ApplicationTemplate.Mobile/ApplicationTemplate.Mobile.csproj:14]: It was the Uno (UWP) before on the Xamarin heads projects. For .NET 6 we use Uno.WinUI variant.

File: src/app/ApplicationTemplate.Presentation/Configuration/AppServicesConfiguration.cs [hunk 2]
```diff
@@ -1,18 +1,10 @@
...
-using ApplicationTemplate;
+﻿using System.Reactive.Concurrency;
 using ApplicationTemplate.Business;
 using ApplicationTemplate.Client;
 using ApplicationTemplate.Presentation;
 using MessageDialogService;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Localization;
-using Microsoft.Extensions.Logging;
+using Xamarin.Essentials;
```
Comment (vincentcastagna) [inline src/app/ApplicationTemplate.Presentation/Configuration/AppServicesConfiguration.cs:7]: Do we still need those namespaces as you seem to not using X.E anymore ?

File: src/app/ApplicationTemplate.Presentation/Framework/Connectivity/IConnectivity.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace ApplicationTemplate;
+
+public interface IConnectivity
```
Comment (takla21) [inline src/app/ApplicationTemplate.Presentation/Framework/Connectivity/IConnectivity.cs:5]: add xml docs

File: src/app/ApplicationTemplate.Presentation/Configuration/AppServicesConfiguration.cs (hunk 2 above)
Comment (Soap-141) [reply src/app/ApplicationTemplate.Presentation/Configuration/AppServicesConfiguration.cs:7]: Currently removing those reference, should be done in the next commit.

File: src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/Connectivity.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+namespace ApplicationTemplate;
+
+public sealed class Connectivity : IConnectivity
+{
+	public Connectivity()
+	{
+		NetworkInformation.NetworkStatusChanged += OnNetworkStatusChanged;
+	}
+
+	~Connectivity()
```
Comment (takla21) [inline src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/Connectivity.cs:15]: I would propose to implement `IDisposable` to unsubscribe to follow the dotnet disposable pattern.

Comment (Soap-141) [reply src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/Connectivity.cs:15]: Good idea.

File: src/app/ApplicationTemplate.Presentation/Framework/Connectivity/IConnectivity.cs (hunk 3 above)
Comment (takla21) [reply src/app/ApplicationTemplate.Presentation/Framework/Connectivity/IConnectivity.cs:8]: also for the property/event

File: src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/Connectivity.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+using Uno.Logging;
+using Windows.Networking.Connectivity;
+
+namespace ApplicationTemplate;
+
+public sealed class Connectivity : IConnectivity, IDisposable
+{
+	public Connectivity()
+	{
+		NetworkInformation.NetworkStatusChanged += OnNetworkStatusChanged;
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/Connectivity.cs:12]: You could probably avoid subscribing until someone subscribes to `ConnectivityChanged`.
This can be refactored in a later PR.

File: src/app/ApplicationTemplate.Shared.Views/Configuration/ApiConfiguration.cs [hunk 6]
```diff
@@ -42,14 +39,6 @@ private static IServiceCollection AddMainHandler(this IServiceCollection service
...
-//-:cnd:noEmit
-#if WINDOWS_UWP || __ANDROID__ || __IOS__
-		// TODO #172362: Not implemented in Uno.
-		// return NetworkInformation.GetInternetConnectionProfile()?.GetNetworkConnectivityLevel() == NetworkConnectivityLevel.InternetAccess;
-		return Task.FromResult(Xamarin.Essentials.Connectivity.NetworkAccess == Xamarin.Essentials.NetworkAccess.Internet);
-#else
-		return Task.FromResult(true);
-#endif
-//+:cnd:noEmit
+		return Task.FromResult(NetworkInformation.GetInternetConnectionProfile()?.GetNetworkConnectivityLevel() == NetworkConnectivityLevel.InternetAccess);
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Shared.Views/Configuration/ApiConfiguration.cs:42]: shouldn't this use your newly added `IConnectivity`? That way it could be mocked.

File: src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/ConnectivityProvider.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+using Uno.Logging;
+using Windows.Networking.Connectivity;
+
+namespace ApplicationTemplate;
+
+public sealed class ConnectivityProvider : IConnectivityProvider, IDisposable
+{
+	private bool _subscribed = false;
+
+	private EventHandler<ConnectivityChangedEventArgs> _connectivityChanged;
```
[the remark this answers is not in the document]
Comment (Soap-141) [reply src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/ConnectivityProvider.cs:12]: Well the event is the public one, I don't know, I have to see if I can find some documentation.

[the remark this answers is not in the document]
Comment (Soap-141) [reply src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/ConnectivityProvider.cs:12]: Looks good. https://stackoverflow.com/questions/1015166/c-event-with-explicity-add-remove-typical-event

[the remark this answers is not in the document]
Comment (Soap-141) [reply src/app/ApplicationTemplate.Shared.Views/Framework/Connectivity/ConnectivityProvider.cs:12]: Both should be an event you think? I only added the private field to be able to invoke the handler.

**Review by Marc-Antoine-Soucy — APPROVED** [review]

**Review by jeanplevesque — APPROVED** [review]
